device_test.go

```diff
@@ -227,20 +217,54 @@ func (*failDetachLib) detachKernelDriver(h *libusbDevHandle, i uint8) error {
 
 func TestAutoDetachFailure(t *testing.T) {
 	t.Parallel()
-	fake := newFakeLibusb()
-	c := newContextWithImpl(&failDetachLib{fake})
-	defer c.Close()
-	dev, err := c.OpenDeviceWithVIDPID(0x8888, 0x0002)
-	if dev == nil {
-		t.Fatal("OpenDeviceWithVIDPID(0x8888, 0x0002): got nil device, need non-nil")
-	}
-	defer dev.Close()
+	dev, err := createFakeDevice(0x8888, 0x0002)
 	if err != nil {
-		t.Fatalf("OpenDeviceWithVIDPID(0x8888, 0x0002): %v", err)
+		t.Error(err)
 	}
 	dev.SetAutoDetach(true)
 	_, err = dev.Config(1)
 	if err == nil {
 		t.Fatalf("%s.Config(1) got nil, but want no nil because interface fails to detach", dev)
 	}
 }
+
+func TestActiveConfigNumFailure(t *testing.T) {
+	dev, err := createFakeDevice(0x8888, 0x0002)
+	if err != nil {
+		t.Error(err)
+	}
+
+	dev.handle = nil
+	_, err = dev.ActiveConfigNum()
+	if err == nil {
+		t.Fatalf("Error on failing ActiveConfigNum()")
+	}
+}
+
+func TestDeviceDescString(t *testing.T) {
```

Comment:
this is a bad practice, given that the strings are not intended for machine consumption. It leads to so-called "change detection tests", where non-meaningful changes in the code (like the change of the display string format) require matching and very mechanical updates in tests. At the same time such tests prevent very few meaningful bugs. The net result that it costs more to maintain the test than it saves on reliability.
I'd rather not test the verbatim strings.